src/transformers/tokenization_utils.py

```diff
@@ -1495,36 +1552,59 @@ def _convert_encoding(
         return_attention_mask=True,
         return_overflowing_tokens=False,
         return_special_tokens_mask=False,
+        return_offsets_mapping=False,
+        pad_token_id: int = 0,
+        pad_to_length: int = -1,
     ):
+        if return_overflowing_tokens and encoding.overflowing is not None:
+            encodings = [encoding] + encoding.overflowing
+        else:
+            encodings = [encoding]
+
         encoding_dict = {
-            "input_ids": encoding.ids,
+            "input_ids": [e.ids for e in encodings],
         }
+
         if return_token_type_ids:
-            encoding_dict["token_type_ids"] = encoding.type_ids
+            encoding_dict["token_type_ids"] = [e.type_ids for e in encodings]
         if return_attention_mask:
-            encoding_dict["attention_mask"] = encoding.attention_mask
-        if return_overflowing_tokens:
-            overflowing = encoding.overflowing
-            encoding_dict["overflowing_tokens"] = overflowing.ids if overflowing is not None else []
+            encoding_dict["attention_mask"] = [e.attention_mask for e in encodings]
         if return_special_tokens_mask:
-            encoding_dict["special_tokens_mask"] = encoding.special_tokens_mask
+            encoding_dict["special_tokens_mask"] = [e.special_tokens_mask for e in encodings]
+        if return_offsets_mapping:
```

Comment:
If we don't give access to the normalized string somehow, we should maybe provide offsets to the original string here. Wdyt?

Reply:
Hum, currently the offsets are given w.r.t the normalised string ? If this is the case, then yes we may want to provide offsets in the original string then, or expose an utility method doing the mapping in Python. 

Is it something we can easily expose on `Encoding` ?

Reply:
Yes, offsets are related to the normalized string. You can retrieve the `original` offsets by doing `encoding.original_str.offsets[encoding.offsets[X]]`

Reply:
I haven't followed the discussion very closely, but shouldn't offsets be returned based on the original string by default? 

As an end user I don't think I really care about the normalized (internal) version of my input.

Thoughts?

Reply:
I tend to agree with @julien-c, the normalized string is more an internal representation